### PR TITLE
fix: ignore the case when updating tags

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

The tags on azure are case-insensitive for both keys and values. We need to ignore the case when comparing existing and new tags.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
fix: ignore the case when updating Azure tags
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/triage accepted
/sig cloud-provider
/area provider/azure
/priority critical-urgent
cc @feiskyer

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go [hunk 1]
```diff
@@ -138,3 +138,28 @@ func parseTags(tags string) map[string]*string {
 	}
 	return formatted
 }
+
+func findKeyInMapCaseInsensitive(targetMap map[string]*string, key string) (bool, string) {
```
[the remark this answers is not in the document]
Comment (nilo19) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go:142]: The returned key is the one existing in the tag currently, and it is not sure whether the `k` and `key` are in the same case. So I think it's clear to add another return value.

**Comment by nilo19** [comment]
@cheftako could you please take a look again?

**Comment by ialidzhikov** [comment]
@nilo19 , I see you created backports for 1.20 and 1.21 but backport for 1.22 is missing? Can you please check?

**Comment by nilo19** [comment]
@ialidzhikov yeah, I need https://github.com/kubernetes/kubernetes/pull/104705 to be merged and I will continue the cherry pick.

**Comment by saschagrunert** [comment]
@nilo19 please open all cherry-pick PR's in parallel if possible, that way we can ensure that they merge (almost) simultaneously.

**Pull request merged**